hal/crc_api.h

```diff
@@ -109,6 +105,24 @@ extern "C" {
  *
  * # Potential bugs
  *
+ * # Macros
+ *
+ * Platform support for particular CRC polynomials is indicated by the
+ * macro HAL_CRC_IS_SUPPORTED(polynomial, width), which must be defined
+ * in device.h, or a file included from there.
```

Comment:
In this pr, it is defined in `objects.h`

Reply:
I don't believe `objects.h` is a formal part of the HAL API - the core code includes `device.h`, and most (or all?) targets include `objects.h` from there. 

It seems the usual structure `device.h` is usually maximally specific for a target, and those include a more common `objects.h`.

Reply:
`device.h` is the one. We used to have `objects.h` but after added config system it almost vanished.